test/README.md

```diff
@@ -0,0 +1,103 @@
+# Running Tests
+
+To run all tests, perform the following steps:
+
+## Startup a fresh iRODS server
+
+The iRODS testing environment is assumed to be used to setup an iRODS server, with a minimum version of 4.3.2.
+See the testing environment to setup a clean iRODS server: https://github.com/irods/irods_testing_environment
+
+### Get the iRODS network information
+
+We need the network that the HTTP API is running on to have all pieces of the server
+communicate with each other.
+
+Following is an example of how to list all available docker networks with examples
+of how the output may appear:
+
+```console
+$ docker network ls
+NETWORK ID     NAME                              DRIVER    SCOPE
+d06849108e9c   bridge                            bridge    local
+371117285a13   host                              host      local
+fbf586b6459f   none                              null      local
+297992bf57a9   ubuntu-2204-postgres-14_default   bridge    local
+```
+
+In the example output, the desired network would be `ubuntu-2204-postgres-14_default`,
+which would have been created by the testing environment with a running iRODS server.
+
+The docker network used will be referred to as `<NETWORK>` in the following sections.
+
+## Startup the Keycloak image for testing
+
+To run tests that require an OpenID Provider, make sure you build the image.
+To build the image simply run the following command in the `keycloak` directory:
+
+```console
+docker build -f irods-http-api-keycloak.Dockerfile -t irods-http-api-keycloak .
+```
+
+After which, run the following command to startup keycloak:
+
+```console
+docker run --rm --network <NETWORK> -p 8080:8080 irods-http-api-keycloak start-dev
+```
+
+### Get detailed network information
+
+Detailed information of the network will be required to properly configure the HTTP API.
+For example, `/http_server/host` will need to be set to the correct address given by the network
```

Comment:
How about qualifying the config property so it's clear to everyone?
```
For example, `/http_server/host` _(in the HTTP API's configuration file)_ will ...
```

Reply:
Done.